Sources/Brave/Frontend/Browser/Favorites/FavoritesViewController.swift

```diff
@@ -358,7 +358,15 @@ extension FavoritesViewController: UICollectionViewDelegateFlowLayout {
           let openInNewPrivateTab = UIAction(
             title: Strings.openNewPrivateTabButtonTitle,
             handler: UIAction.deferredActionHandler { _ in
-              self.action(bookmark, .opened(inNewTab: true, switchingToPrivateMode: true))
+              if Preferences.Privacy.privateBrowsingLock.value {
+                self.askForLocalAuthentication { [weak self] success, error in
+                  if success {
+                    self?.action(bookmark, .opened(inNewTab: true, switchingToPrivateMode: true))
+                  }
+                }
+              } else {
+                self.action(bookmark, .opened(inNewTab: true, switchingToPrivateMode: true))
+              }
```

Comment:
This is technically duplicated logic. The action here calls `self?.action(...)` which hits the `handleFavoriteAction` method you added the auth check to already. You can delete either this one or that, this one probably makes more sense to keep since its a direct action

Reply:
Fixed at https://github.com/brave/brave-ios/pull/7959/commits/68dbdac411e43999da1077d4dc34a50e45e5d03c